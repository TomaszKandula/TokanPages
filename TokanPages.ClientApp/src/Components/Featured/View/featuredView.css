.featured-card-image {
    height: 256px !important;
    object-fit: cover;
}

.featured-card-height {
    min-height: 175px;
}
