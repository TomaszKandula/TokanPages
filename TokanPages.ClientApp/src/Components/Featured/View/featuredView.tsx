import * as React from "react";
import Box from "@material-ui/core/Box";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import CardActionArea from "@material-ui/core/CardActionArea";
import Skeleton from "@material-ui/lab/Skeleton";
import { IGetFeaturedContent } from "../../../Redux/States/Content/getFeaturedContentState";
import { renderCardMedia } from "../../../Shared/Components/CustomCardMedia/customCardMedia";
import { FEATURED_IMAGE_PATH } from "../../../Shared/constants";
import { FeaturedStyle } from "./featuredStyle";

export const FeaturedView = (props: IGetFeaturedContent): JSX.Element => 
{
    const classes = FeaturedStyle();
    return (
        <section className={classes.section}>
            <Container maxWidth="lg">
                <Box pt={8} pb={5} textAlign="center">
                    <Typography className={classes.caption_text} data-aos="fade-down">
                        {props.isLoading ? <Skeleton variant="text" /> : props.content?.caption?.toUpperCase()}
                    </Typography>
                </Box>
                <Box pb={15} textAlign="center">
                    <Grid container spacing={6}>
                        <Grid item xs={12} md={4} data-aos="fade-up" data-aos-delay="350">
                            <Card elevation={0} className={classes.card}>
                                <CardActionArea href={props.content?.link1} target="_blank">
                                    {props.isLoading 
                                        ? <Skeleton variant="rect" height="256px" /> 
                                        : renderCardMedia(FEATURED_IMAGE_PATH, props.content?.image1, classes.card_media)}
                                    <CardContent className={classes.card_content}>
                                        <Typography className={classes.card_title}>
                                            {props.isLoading ? <Skeleton variant="text" width="250px" /> : props.content?.title1}
                                        </Typography>
                                        <Typography className={classes.card_subtitle}>
                                            {props.isLoading ? <Skeleton variant="text" width="250px" /> : props.content?.subtitle1} 
                                        </Typography>
                                    </CardContent>
                                </CardActionArea>
                            </Card>
                        </Grid>
                        <Grid item xs={12} md={4} data-aos="fade-up" data-aos-delay="150">
                            <Card elevation={0} className={classes.card}>
                                <CardActionArea href={props.content?.link2} target="_blank">
                                    {props.isLoading 
                                        ? <Skeleton variant="rect" height="256px" /> 
                                        : renderCardMedia(FEATURED_IMAGE_PATH, props.content?.image2, classes.card_media)}
                                    <CardContent className={classes.card_content}>
                                        <Typography className={classes.card_title}>
                                            {props.isLoading ? <Skeleton variant="text" width="250px" /> : props.content?.title2}
                                        </Typography>
                                        <Typography className={classes.card_subtitle}>
                                            {props.isLoading ? <Skeleton variant="text" width="250px" /> : props.content?.subtitle2} 
                                        </Typography>
                                    </CardContent>
                                </CardActionArea>
                            </Card>
                        </Grid>
                        <Grid item xs={12} md={4} data-aos="fade-up" data-aos-delay="550">
                            <Card elevation={0} className={classes.card}>
                                <CardActionArea href={props.content?.link3} target="_blank">
                                    {props.isLoading 
                                        ? <Skeleton variant="rect" height="256px" /> 
                                        : renderCardMedia(FEATURED_IMAGE_PATH, props.content?.image3, classes.card_media)}
                                    <CardContent className={classes.card_content}>
                                        <Typography className={classes.card_title}>
                                            {props.isLoading ? <Skeleton variant="text" width="250px" /> : props.content?.title3}
                                        </Typography>
                                        <Typography className={classes.card_subtitle}>
                                            {props.isLoading ? <Skeleton variant="text" width="250px" /> : props.content?.subtitle3} 
                                        </Typography>
                                    </CardContent>
                                </CardActionArea>
                            </Card>
                        </Grid>
                    </Grid>
                </Box>
            </Container>
        </section>
    );
}
