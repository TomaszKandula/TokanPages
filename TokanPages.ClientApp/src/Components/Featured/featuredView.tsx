import * as React from "react";
import Box from "@material-ui/core/Box";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import Card from "@material-ui/core/Card";
import CardActionArea from "@material-ui/core/CardActionArea";
import CardHeader from "@material-ui/core/CardHeader";
import Skeleton from "@material-ui/lab/Skeleton";
import { IGetFeaturedContent } from "../../Redux/States/Content/getFeaturedContentState";
import { renderCardMedia } from "../../Shared/Components/CustomCardMedia/customCardMedia";
import featuredStyle from "./featuredStyle";

const FeaturedView = (props: IGetFeaturedContent): JSX.Element => 
{
    const classes = featuredStyle();
    return (
        <section className={classes.section}>
            <Container maxWidth="lg">
                <div data-aos="fade-up">
                    <Box pt={8} pb={10}>
                        <Container maxWidth="sm">
                            <Box textAlign="center" mb={5}>
                                <Typography variant="h4" component="h2" gutterBottom={true}>
                                    {props.isLoading ? <Skeleton variant="text" /> : props.content?.caption}
                                </Typography>
                                <Typography variant="subtitle1" color="textSecondary">
                                    {props.isLoading ? <Skeleton variant="text" /> : props.content?.text}
                                </Typography>
                            </Box>
                        </Container>
                        <Container maxWidth="md">
                            <Box textAlign="center">
                                <Grid container spacing={4}>
                                    <Grid item xs={12} md={4}>
                                        <Card elevation={3}>
                                            <CardActionArea href={props.content?.link1} target="_blank">
                                                {props.isLoading 
                                                    ? <Skeleton variant="rect" height="256px" /> 
                                                    : renderCardMedia(props.content?.image1, classes.media)}
                                                <CardHeader 
                                                    title={props.content?.title1} 
                                                    subheader={props.content?.subtitle1} 
                                                    titleTypographyProps={{gutterBottom: true}} 
                                                />
                                            </CardActionArea>
                                        </Card>
                                    </Grid>
                                    <Grid item xs={12} md={4}>
                                        <Card elevation={3}>
                                            <CardActionArea href={props.content?.link2} target="_blank">
                                                {props.isLoading 
                                                    ? <Skeleton variant="rect" height="256px" /> 
                                                    : renderCardMedia(props.content?.image2, classes.media)}
                                                <CardHeader 
                                                    title={props.content?.title2} 
                                                    subheader={props.content?.subtitle2} 
                                                    titleTypographyProps={{gutterBottom: true}} 
                                                />
                                            </CardActionArea>
                                        </Card>
                                    </Grid>
                                    <Grid item xs={12} md={4}>
                                        <Card elevation={3}>
                                            <CardActionArea href={props.content?.link3} target="_blank">
                                                {props.isLoading 
                                                    ? <Skeleton variant="rect" height="256px" /> 
                                                    : renderCardMedia(props.content?.image3, classes.media)}
                                                <CardHeader 
                                                    title={props.content?.title3} 
                                                    subheader={props.content?.subtitle3} 
                                                    titleTypographyProps={{gutterBottom: true}}
                                                />
                                            </CardActionArea>
                                        </Card>
                                    </Grid>
                                </Grid>
                            </Box>
                        </Container>
                    </Box>
                </div>
            </Container>
        </section>
    );
}

export default FeaturedView;
