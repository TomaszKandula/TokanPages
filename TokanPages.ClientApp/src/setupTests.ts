// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
