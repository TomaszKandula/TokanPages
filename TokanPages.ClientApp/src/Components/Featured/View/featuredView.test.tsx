import "../../../setupTests";
import React from "react";
import * as Redux from "react-redux";
import { render } from "enzyme";
import { FeaturedView } from "./featuredView";

jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useSelector: jest.fn(),
}));

describe("test component: featuredView", () => {
    const testContent = {
        language: "eng",
        caption: "Featured",
        text: "I picked three articles that I wrote...",
        title1: "Memory Management",
        subtitle1: "The basics you should know, when writing an application...",
        link1: "https://.../net-memory-management-64389b01f642",
        image1: "article0.jpg",
        title2: "Stored Procedures",
        subtitle2: "I explain why I do not need them that much...",
        link2: "https://.../i-said-goodbye-to-stored-procedures-539d56350486",
        image2: "article1.jpg",
        title3: "SQL Injection",
        subtitle3: "This article will explore the issue in greater detail...",
        link3: "https://.../sql-injection-1bde8bb76ebc",
        image3: "article2.jpg",
    };

    beforeEach(() => {
        jest.spyOn(Redux, "useSelector").mockReturnValueOnce({
            isLoading: false,
            content: testContent,
        });
    });

    it("should render correctly '<FeaturedView />' when content is loaded.", () => {
        const html = render(<FeaturedView />);
        expect(html).toMatchSnapshot();
    });
});
