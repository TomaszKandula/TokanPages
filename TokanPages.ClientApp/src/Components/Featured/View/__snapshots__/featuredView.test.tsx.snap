// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component: featuredView should render correctly '<FeaturedView />' when content is loaded. 1`] = `
<section
  class="makeStyles-section-1"
>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      class="MuiBox-root MuiBox-root-8"
    >
      <p
        class="MuiTypography-root makeStyles-caption_text-2 MuiTypography-body1"
        data-aos="fade-down"
      >
        FEATURED
      </p>
    </div>
    <div
      class="MuiBox-root MuiBox-root-9"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-6"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
          data-aos="fade-up"
          data-aos-delay="350"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-card-3 MuiPaper-elevation0 MuiPaper-rounded"
          >
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiCardActionArea-root"
              href="https://.../net-memory-management-64389b01f642"
              tabindex="0"
              target="_blank"
            >
              <div
                class="MuiCardMedia-root makeStyles-card_media-7"
                style="background-image:url(\\"http://localhost:5000/api/v1/content/assets/getNonVideoAsset/?blobName=images/sections/featured/article0.jpg\\")"
              />
              <div
                class="MuiCardContent-root makeStyles-card_content-4"
              >
                <p
                  class="MuiTypography-root makeStyles-card_title-5 MuiTypography-body1"
                >
                  Memory Management
                </p>
                <p
                  class="MuiTypography-root makeStyles-card_subtitle-6 MuiTypography-body1"
                >
                  The basics you should know, when writing an application...
                </p>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
          data-aos="fade-up"
          data-aos-delay="150"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-card-3 MuiPaper-elevation0 MuiPaper-rounded"
          >
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiCardActionArea-root"
              href="https://.../i-said-goodbye-to-stored-procedures-539d56350486"
              tabindex="0"
              target="_blank"
            >
              <div
                class="MuiCardMedia-root makeStyles-card_media-7"
                style="background-image:url(\\"http://localhost:5000/api/v1/content/assets/getNonVideoAsset/?blobName=images/sections/featured/article1.jpg\\")"
              />
              <div
                class="MuiCardContent-root makeStyles-card_content-4"
              >
                <p
                  class="MuiTypography-root makeStyles-card_title-5 MuiTypography-body1"
                >
                  Stored Procedures
                </p>
                <p
                  class="MuiTypography-root makeStyles-card_subtitle-6 MuiTypography-body1"
                >
                  I explain why I do not need them that much...
                </p>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight"
              />
            </a>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
          data-aos="fade-up"
          data-aos-delay="550"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-card-3 MuiPaper-elevation0 MuiPaper-rounded"
          >
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiCardActionArea-root"
              href="https://.../sql-injection-1bde8bb76ebc"
              tabindex="0"
              target="_blank"
            >
              <div
                class="MuiCardMedia-root makeStyles-card_media-7"
                style="background-image:url(\\"http://localhost:5000/api/v1/content/assets/getNonVideoAsset/?blobName=images/sections/featured/article2.jpg\\")"
              />
              <div
                class="MuiCardContent-root makeStyles-card_content-4"
              >
                <p
                  class="MuiTypography-root makeStyles-card_title-5 MuiTypography-body1"
                >
                  SQL Injection
                </p>
                <p
                  class="MuiTypography-root makeStyles-card_subtitle-6 MuiTypography-body1"
                >
                  This article will explore the issue in greater detail...
                </p>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
